packages/react-google-maps-api/src/utils/injectscript.ts

```diff
@@ -16,21 +16,30 @@ export const injectScript = ({ url, id }: InjectScriptArg): Promise<any> => {
 
   return new Promise(function injectScriptCallback(resolve, reject) {
     const existingScript = document.getElementById(id) as HTMLScriptElement | undefined
+    const windowWithGoogleMap: WindowWithGoogleMap = window
     if (existingScript) {
       // Same script id/url: keep same script
       if (existingScript.src === url) {
-        return resolve(id)
+        if(existingScript.getAttribute('data-state') === 'ready') {
+          return resolve(id)
+        }
+        else {
+          const onGoogleMapsReadyClosure = windowWithGoogleMap.initMap
```

Comment:
It's kinda hard to understand what's going on here. Had to read it several times to understand it's actually calling original version of that function to resolve all promises in the row. Perhaps even some comment would be useful.
```suggestion
          const originalInitMap = windowWithGoogleMap.initMap
```